Story block: Make stories available in the block inserter

Tug opened the pull request:
#### Changes proposed in this Pull Request:
This is the release PR for the story block. It just flips the switch and makes the block available from the inserter in production.

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* Run `yarn build-production`
* Load the editor, make sure the story block is available from the inserter and works as expected

aforcier reviewed (dismissed):
Working great! Tested with a production Jetpack build and on WP.com. :shipit: 🎉

Tug commented on projects/plugins/jetpack/extensions/blocks/story/index.js:
```diff
@@ -58,7 +57,7 @@ export const settings = {
 	supports: {
 		html: false,
-		inserter: process.env.NODE_ENV !== 'production' || Platform.OS !== 'web', // false for web, true for mobile
+		inserter: true,
```
👍

Tug commented:
I have addressed your feedback @jeherve. Btw let's hold off the merge for a bit on this one as I'd like https://github.com/Automattic/jetpack/pull/19876 to land first.

jeherve reviewed (approved):
Ready to merge whenever you are ready to push the button.

Tug commented:
As discussed on other channels with @aforcier let's not wait for #19876 which will need extensive testing before it gets merged.

Tug commented:
Deployed to WPCOM with changeset ID 226118
